lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -486,6 +488,44 @@ _LoadTexture(const std::string& path, bool& isColorSpaceSRGB, MFloatArray& uvSca
         desc.fFormat = MHWRender::kR32G32B32_FLOAT;
         texture = textureMgr->acquireTexture(path.c_str(), desc, spec.data);
         break;
+    case HioFormatFloat16Vec3: {
+        // R16G16B16 is not supported by VP2. Converted to R16G16B16A16.
+        constexpr int bpp_8 = 8;
+
+        desc.fFormat = MHWRender::kR16G16B16A16_FLOAT;
+        desc.fBytesPerRow = spec.width * bpp_8;
+        desc.fBytesPerSlice = desc.fBytesPerRow * spec.height;
+
+        GfHalf               opaqueAlpha(1.0f);
+        const unsigned short alphaBits = opaqueAlpha.bits();
+        const unsigned char  lowAlpha = reinterpret_cast<const unsigned char*>(&alphaBits)[0];
+        const unsigned char  highAlpha = reinterpret_cast<const unsigned char*>(&alphaBits)[1];
+
+        std::vector<unsigned char> texels(desc.fBytesPerSlice);
+
+        tbb::parallel_for(tbb::blocked_range<int>(0, spec.height), [&](tbb::blocked_range<int> r) {
```

Comment:
The default grain size is 1. It might be too small depending on the width. Could it be better to use `blocked_range2d`?

Reply:
I thought it would be better to only parallelize in the Y direction due to data access, dividing in X would reduce locality. (And there are enough values in height to allow parallel work.)

The default partitioner (auto_partitioner) implement the automatic grainsize heuristic, so I don't think we need to specify a grain size. Given the height of the image will be much larger than the number of CPU,

Reply:
Ok, makes sense. Thanks for elaboration. Did you profile the loop performance before and after the change?

Reply:
No.

Reply:
Could you profile it for a large texture and a small texture, just to be sure about the change?

Reply:
I'm currently rebuilding USD 20.02, I'll look at profiling when I've done the old GL versions of conversion.

Reply:
I profiled with a 18MB texture. The parallel version is slower, but in both cases, the cost of conversion is minimal vs the I/O and rendering, etc. So It does not matter much either way. IDK if 18MB is considered large or not, maybe the parallel version is more "future proof".

Reply:
What is the resolution of the 18MB texture?

Reply:
2048x2048

Reply:
2kx2k is common. Maybe try 4kx4k and we can see some performance gain?

Reply:
I do agree with you the major overhead here would be I/O overhead and VRAM transfer overhead. So it might not make a big difference overall.

Reply:
I like @HdC-adsk suggestions to do a scalability test with a few different sizes of the texture. If we can't prove with the data that there is a value in doing the parallel version, why use up all the resources.

Additionally, is _LoadTexture called from main-thread on it is part of multithreaded sync execution?

Reply:
@kxl-adsk Currently all sprim syncs (from which _LoadTexture is called) are serial on main thread, however, in the future it might be changed to multithreaded sync according to the post below:

https://groups.google.com/g/usd-interest/c/WH6DgLDwNvE/m/7ngJnkYoCAAJ

Reply:
Thx, so we have additional resources currently available (since there is nothing else running at that time). Still would be good to do the scalability test. @pierrebai-adsk do you agree?

Reply:
I can try more textures. Do we have bigger EXR files I can test with? Otherwise I'll try to search.

Honestly, this may all fall into the "premature optimization" bucket, we could revert to the simple loop until we do have data showing it is a bottleneck. (The only data point so far with that 18MB texture pointing toward being detrimental.)

Reply:
The image content doesn't matter, so I guess we can easily create some white images with different resolutions and convert them to EXR.

If all resolutions cannot benefit from the concurrent loop, I would agree with the simple loop.

Reply:
👍

Reply:
I converted a 4800x6500 image in EXR 16-bits, resulting in a 186MB image. It was faster without parallel for, In all cases, the total time to load / convert /etc was 8% with 5% being spent simply on I/O (in image->Read(spec) at line 461), on a reasonable SSD. So, the I/O is still the biggest factor anyway.

Reply:
Perfect. Thank you for providing this data.